### PR TITLE
bug catching logger

### PR DESCRIPTION
**Pull request by j-baker**

### CODE REVIEW

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java [hunk 1]
```diff
@@ -43,16 +46,27 @@
...
+                            uncaughtException))
+            .setNameFormat("remoting-okhttp-dispatcher-%d")
+            .build();
     /**
      * The {@link ExecutorService} used for the {@link Dispatcher}s of all OkHttp clients created through this class.
+     * Same as OkHttp's default, but with a logging uncaught exception handler.
      */
-    private static final ExecutorService executionExecutor = Tracers.wrap(new Dispatcher().executorService());
+    private static final ExecutorService executionExecutor =
+            Tracers.wrap(Executors.newCachedThreadPool(executionThreads));
```
Comment (uschi2000) [inline okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java:69]: should we use the same type of executor as the okhttp Dispatcher?

Comment (j-baker) [reply okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java:69]: it is!

Comment (j-baker) [reply okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java:69]: noted in the Javadoc

Comment (uschi2000) [reply okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java:69]: ah, just had to chase the `Executors` call. thanks.

**Review by uschi2000 — APPROVED** [review]

**Pull request merged**